control_plane/src/compute.rs

```diff
@@ -152,7 +150,6 @@ impl PostgresNode {
         let port: u16 = conf.parse_field("port", &context)?;
         let timeline_id: TimelineId = conf.parse_field("neon.timeline_id", &context)?;
         let tenant_id: TenantId = conf.parse_field("neon.tenant_id", &context)?;
-        let uses_wal_proposer = conf.get("neon.safekeepers").is_some();
```

Comment:
Do I understand correctly that:

1. flakyness was caused by inherent race of pageserver not having the truly latest lsn because on shutdown safekeepers may have had more recent lsn
2. in the test we always called `self.load_basebackup(None)` path from `PostgresNode::load_basebackup`
    - this was failing sometimes because of (1)

Now with this fix in the test we will always sync with the safekeepers first and then use that lsn for `self.do_basebackup(lsn)`.

(I assume sync_safekeepers negotiates the latest lsn with safekeeper(s) in test and that will always have prev lsn because we will wait for that lsn to arrive on pageserver and everything will be ok.)

Reply:
Discussed this in a call, my understanding was somewhat correct.